test/test_game.py

```diff
@@ -0,0 +1,229 @@
+import unittest
+
+from rock_paper_scissors.core.exc import InvalidRoundsError, InvalidChoiceError
+from rock_paper_scissors.core.game import Game
+from rock_paper_scissors.core.moves import Paper, Rock
+from rock_paper_scissors.core.user import User
+from rock_paper_scissors.core.valid_choices import VALID_CHOICES
+
+
+class GameInitTestCase(unittest.TestCase):
+    def test_player1_and_player2_and_starting_rounds_types(self):
+        player1 = 'Joe'
+        player2 = 'Jane'
+        game = Game(player1, player2)
+
+        self.assertIsInstance(game.player1, str)
+        self.assertIsInstance(game.player2, str)
+        self.assertEqual(game._rounds, 0)
+
+
+class GameSetRoundsTestCase(unittest.TestCase):
+    def setUp(self):
+        self.game = Game('player1', 'player2')
+
+    def test_valid_rounds_input(self):
+        value = self.game.rounds = 11
+
+        self.assertEqual(value, 11)
```

Comment:
I would say not, because anyone reading the tests can see for themselves that it's odd.

If it were less obvious, you could show some kind of proof that the value is odd, like:
```python
valid_value = 11
self.assertEqual(
    11 % 2, 
    1, 
    'Precondition failed: `valid_value` is not odd and is therefore invalid'
)

value = self.game.rounds = 11

self.assertEqual(value, 11)
```
Which you could certainly do in this case, but I still wouldn't call it "necessary" :smile:

Reply:
Gotchya. The `assertRaises` method also asserts that the value is odd.